### PR TITLE
internal/store/validatingwebhookconfiguration.go: Switch to v1

### PR DESCRIPTION
**Pull request by lilic**
Initial fix was done https://github.com/kubernetes/kube-state-metrics/pull/1043/files but it seems like it didn't adjust all of it

As we are expecting v1 we should also create a watch on v1. Otherwise
see the following error in the logs:

> expected type *v1.ValidatingWebhookConfiguration, but watch event object had type *v1beta1.ValidatingWebhookConfiguration

cc @brancz @tariq1890

### CODE REVIEW

**Comment by tariq1890** [comment]
Thanks @LiliC let’s go with v1 completely

**Comment by tariq1890** [comment]
@lilic just following up on this :)

**Comment by LiliC** [comment]
yeah tests failed looking at this.

**Comment by lilic** [comment]
So strange, I ran this locally and did not seen any problems 🤔

**Comment by lilic** [comment]
Figured out the problem, seems like we had some manifests v1beta1 and also old k8s cluster.

**Comment by lilic** [comment]
@tariq1890 PTAL, tests passed now, thanks!

File: tests/e2e.sh [hunk 1]
```diff
@@ -17,7 +17,7 @@
 set -e
 set -o pipefail
 
-KUBERNETES_VERSION=v1.15.3
+KUBERNETES_VERSION=v1.17.0
```
Comment (tariq1890) [inline tests/e2e.sh]:
```suggestion
KUBERNETES_VERSION=v1.17.3
```

**Comment by tariq1890** [comment]
Looks good. Just one comment

**Comment by tariq1890** [comment]
@lilic Also, don't forget to squash the commit history :)

**Comment by lilic** [comment]
Done, PTAL @tariq1890

**Comment by lawliet89** [comment]
On Kubernetes `1.15.8-gke.3`, after upgrading, I get endless `E0221 04:47:38.447162       1 reflector.go:156] pkg/mod/k8s.io/client-go@v0.0.0-20191109102209-3c0d1af94be5/tools/cache/reflector.go:108: Failed to list *v1.ValidatingWebhookConfiguration: the server could not find the requested resource`.

**Comment by filip-havlicek** [comment]
@lawliet89 I have the same problem
it's because there is no admissionregistration.k8s.io/v1 in Kubernetes 1.15, there is only admissionregistration.k8s.io/v1beta1 (ValidatingWebhookConfiguration is part of admissionregistration.k8s.io api group)

**Comment by lilic** [comment]
@lawliet89 @filip-havlicek you can disable that resource via args to kube-state-metrics or switch to using older version. We will be updating our compatibility matrix to reflect this, thanks for reporting.

Note that this change did not introduce this switch, this happened in an earlier PR, but only partially. So this just fixed that bug.

**Pull request merged**